nodejs/aws-infra/experimental/fargate.ts

```diff
@@ -0,0 +1,198 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as aws from "@pulumi/aws";
+import * as pulumi from "@pulumi/pulumi";
+
+import * as utils from "../utils";
+
+import * as mod from ".";
+
+export type FargateTaskDefinitionArgs = utils.Overwrite<mod.TaskDefinitionArgs, {
```

Comment:
Seems a little strange to use a codename like "Fargate" in the class definitions.

Reply:
I actually think this is pretty reasonable.  This is the TaskDefintion you use when you know you're targetting fargate, and thus you don't need to supply a bunch of dummy defaults we can take care of for you.

Reply:
Do you have an alternate approach you think would be nicer?